vector/src/main/java/im/vector/app/features/pin/lockscreen/ui/LockScreenViewModel.kt

```diff
@@ -181,8 +176,19 @@ class LockScreenViewModel @AssistedInject constructor(
         }
     }
 
-    private fun updateConfiguration(configuration: LockScreenConfiguration) {
-        setState { copy(lockScreenConfiguration = configuration) }
-        updateStateWithBiometricInfo()
+    /**
+     * Wait until the device is unlocked. There seems to be a behavior change on Android 12 that makes [KeyguardManager.isDeviceLocked] return `false` even
+     * after an Activity's `onResume` method. If we mix that with the system keys needing the device to be unlocked before they're used, we get crashes.
+     * See issue [#6768](https://github.com/vector-im/element-android/issues/6768).
+     */
+    @SuppressLint("NewApi")
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal suspend fun waitUntilKeyguardIsUnlocked() {
+        if (versionProvider.get() < Build.VERSION_CODES.S) return
+        withTimeoutOrNull(5.seconds) {
+            while (keyguardManager.isDeviceLocked) {
+                delay(50.milliseconds)
+            }
+        }
```

Comment:
This is the actual code that fixes the issue.

Reply:
do you think it's worth extracting this to a dedicated class/usecase? would also mean we could avoid the `VisibleForTesting`

not a blocker for me

Reply:
Actually, the `@VisibleForTesting` annotation is no longer needed, I forgot to revert it to private.

I also thought about making it an external component, but it seemed too specific since it's only used here and it's a terrible hack that shouldn't be used anywhere if there's an alternative.

Reply:
I don't have tooo strong of an opinion here, up to you! 
I would lean towards extracting as `ViewModels` tend to collect logic, requiring more setup for unit tests